tools/SetupFlow/DevHome.SetupFlow/Models/CloningInformation.cs

```diff
@@ -83,33 +79,76 @@ public bool CloneToExistingDevDrive
     [ObservableProperty]
     private BitmapImage _repositoryTypeIcon;
 
+    /// <summary>
+    /// Sets RepositoryTypeIcon according to the theme.
+    /// </summary>
+    /// <param name="theme">The theme to use to determine the icon.</param>
+    /// <remarks>
+    /// This is for monochrome icons only.
+    /// Additionally, this assumes the icon coming from the extension is colored for the dark theme.
+    /// If the provider is null (Generic repo) or if the icon is null, the git icon is used instead.
+    /// </remarks>
     public void SetIcon(ElementTheme theme)
     {
-        if (theme == ElementTheme.Dark)
+        // RepositoryProvider can be null in the case of URL cloning.
+        if (RepositoryProvider == null || RepositoryProvider.Icon == null)
         {
-            // Currently the only providers are Github and the generic provider.  The provider type
-            // for the generic provider is git.
-            // TODO: Remove when extensions have a GetIcon method.
-            if (RepositoryProviderDisplayName.Equals("github", StringComparison.OrdinalIgnoreCase))
+            BitmapImage gitIcon;
+            if (theme == ElementTheme.Dark)
             {
-                RepositoryTypeIcon = DarkGithub;
+                gitIcon = DarkGit;
             }
             else
             {
-                RepositoryTypeIcon = DarkGit;
+                gitIcon = LightGit;
             }
+
+            RepositoryTypeIcon = gitIcon;
+            return;
         }
-        else
+
+        BitmapDecoder decoder = BitmapDecoder.CreateAsync(RepositoryProvider.Icon.OpenReadAsync().AsTask().Result).AsTask().Result;
+
+        // Get the pixel data as a byte array
+        PixelDataProvider pixelData = decoder.GetPixelDataAsync().AsTask().Result;
+        var pixels = pixelData.DetachPixelData();
+
+        // Assume that the icon is colored for dark mode.

```

Comment:
What happens if the icon is expected to have black inside of it?  Like if I have a smiley face icon with black eyes/smile, I wouldn't want the eyes/smile to change.

Reply:
This changes all black pixels to white and all white to black.  If the background is black, the face is white, and the eyes are black then, when in light mode the background would be white, face black, and eyes white.

The smile/eyes would still be visible, just different colors.  If we can get a icon like this I can test with it.